# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/antitracking/bouncetrackingprotection/BounceTrackingStorageObserver.h
# Commit: 087b27593538
# Full Hash: 087b27593538ac10d0c69e41100423f430ad8c89
# Author: Paul Zuehlcke <pbz@mozilla.com>
# Date: 2023-09-07 21:34:02
# Regressor Bug: 1839918
# File Overlap Count: 1
# Description:
#   Bug 1839918 - Add BounceTrackingStorageObserver to observe cookie changes. r=bvandersloot,anti-tracking-reviewers
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D185304
# ==============================================================================

diff -r 302d12c12083 -r 087b27593538 toolkit/components/antitracking/bouncetrackingprotection/BounceTrackingStorageObserver.h
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/toolkit/components/antitracking/bouncetrackingprotection/BounceTrackingStorageObserver.h	Thu Sep 07 08:25:56 2023 +0000
@@ -0,0 +1,28 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+#ifndef mozilla_BounceTrackingStorageObserver_h__
+#define mozilla_BounceTrackingStorageObserver_h__
+
+#include "mozilla/Logging.h"
+#include "nsIObserver.h"
+
+namespace mozilla {
+
+extern LazyLogModule gBounceTrackingProtectionLog;
+
+class BounceTrackingStorageObserver final : public nsIObserver {
+  NS_DECL_ISUPPORTS
+  NS_DECL_NSIOBSERVER
+
+ public:
+  BounceTrackingStorageObserver() = default;
+  nsresult Init();
+
+ private:
+  ~BounceTrackingStorageObserver() = default;
+};
+
+}  // namespace mozilla
+
+#endif